server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,13 +109,21 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
                 .date(message.getInternalDateAsZonedDateTime())
-                .textBody(messageContent.getTextBody().orElse(null))
-                .htmlBody(messageContent.getHtmlBody().orElse(null))
-                .preview(getPreview(messageContent))
+                .textBody(textBody)
```

Comment:
Can you make it more 'fin grained'? Add more tests:
 - Focussing only the GetMessages method (because mails can be appended via SMTP)
 - And without attachment